clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientBuilderImpl.java

```diff
@@ -212,11 +219,26 @@ public ZeebeClientBuilder credentialsProvider(final CredentialsProvider credenti
 
   @Override
   public ZeebeClient build() {
+    applyOverrides();
     applyDefaults();
 
     return new ZeebeClientImpl(this);
   }
 
+  private void applyOverrides() {
+    if (!insecureConnSetAsProperty && Environment.system().isDefined(PLAINTEXT_CONNECTION_VAR)) {
```

Comment:
The priority is then properties > environment > builder arguments? Shouldn't environment be the top priority?

Reply:
I thought properties should be higher priority because they're specific to the process while environment variables aren't. It was meant to allow changing parameters for one client while leaving others with the generic config. But I think it's mostly a question of use cases, what did you have in mind?

Reply:
The use case for environment variables is that we can package our application in a static way, and overwrite the configuration with environment variables. In my opinion this means they should have higher priority than properties, but I'm not a Java dev and never use property files, so... @menski ?